Binary_Search/Binary_Search.cpp

```diff
@@ -62,42 +62,44 @@ int Upper_Bound(int array[], int size, int desired)
 // Driver Function
 int main()
 {
-    int array[] = {1, 2, 3, 4, 5, 5, 7};
-
-    // Element 4 to be searched
-    if (Binary_Search(array, 7, 4) != -1)
-        cout << "Found" << endl;
-    else
-        cout << "Not Found" << endl;
-
-    //Element 9 to be searched
-    if (Binary_Search(array, 7, 9) != -1)
+    int num;
+    scanf("%d", &num);
+    int array[num];
+    for (int i = 0; i < num; i++) {
+        scanf("%d", &array[i]);
+    }
+    int desired;
+    scanf("%d", &desired);
+    if (Binary_Search(array, num, desired) != -1)
```

Comment:
please change the function name to binarySearch. The above kind of naming is generally followed in c#